### DISCUSSION TITLE
[Core Protocol] op-proposer will be removed in the future?

### DISCUSSION THREAD

**Discussion by dapp-whisperer** [ethereum-optimism/developers#252] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi everyone, been a joy digging into the stack.

In the docs for the **Proposer**, it states:
> “It's likely that the Proposer will be removed in the future, but for now it's a necessary component of the OP Stack”

However it isn’t clear to me how the function of publishing L2 state roots to L1 will be achieved, or how it will no longer be necessary, in the future. Apologies if I've missed it.

Can someone point me to the roadmap or justification of this change?

**Comment by opfocus** [comment]
Currently, I haven't seen any information about it on the public roadmap or plans. You haven't missed any relevant information, so please stay tuned. It will be added to this thread when there is public information available.

**Reply by sbvegan** [reply, marked as answer]
@opfocus is correct, this might just be poorly worded. With our upcoming fault proof improvements, it will allow users to self propose output roots, but there are no plans to remove the op-proposer at the moment.
